R/defineModel.R

```diff
@@ -1,18 +1,115 @@
+                        converge=0.001, deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"),
+                        use.letters=FALSE, allowAllScoresEverywhere = TRUE, guessMat = NULL,
+                        est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL,
+                        slopeMatItemCol=NULL, slopeMatValueCol=NULL, progress = FALSE,
+                        Msteps = NULL, increment.factor=1 , fac.oldxsi=0,
+                        export = list(logfile = TRUE, systemfile = FALSE, history = TRUE,
+                                      covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) ){
+
+### checking/asserting the arguments -------------------------------------------
+  # dat
+  ismc <- attr(dat, "multicore")
+  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+
+  # list: splitted Models
+  checkmate::assert_list(splittedModels, null.ok = TRUE)
+  # subset: irtmodel
+  irtmodel <- match.arg(arg = irtmodel, choices = c("1PL", "2PL", "PCM", "PCM2","RSM",
+                                                    "GPCM", "2PL.groups", "GPCM.design", "3PL"))
+  # data frame: qMatrix
+  checkQmatrixConsistency(qMatrix)
+
+  #' assert vector:
+  #' HG.var, group.var, weight.var, schooltype.var
+  lapply(c(HG.var, group.var, weight.var), checkmate::assert_vector, null.ok = TRUE)
+  checkmate::assert_vector(schooltype.var, len = 1, null.ok = TRUE)
+
+  #' assert numeric:
+  #' minNperItem, boundary, n.plausible, seed, n.iterations, nodes, converge,
+  #' deviancechange, Msteps
+  lapply(c(minNperItem, boundary, n.plausible, n.iterations, nodes, Msteps),
+         checkmate::assert_numeric, len = 1, lower = 0)
+  lapply(c(converge, deviancechange), checkmate::assert_numeric, len = 1)
+  checkmate::assert_numeric(seed, len = 1, null.ok = TRUE)
+
+  #' assert character via match.arg:
+  # software
   software <- match.arg(arg = software, choices = c("conquest","tam"))
-  if ( software == "conquest" && is.null(conquest.folder) ) {
+  if(software == "conquest" && is.null(conquest.folder)) {
     conquest.folder <- identifyConquestFolder()
   }
-  ismc <- attr(dat, "multicore")
-  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+  # method
+  method <- match.arg(arg = method, choices = c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"))
+
+  #' assert logical:
+  lapply(c(check.for.linking, removeMinNperItem, remove.boundary, remove.no.answers,
+           remove.no.answersHG, remove.missing.items, remove.constant.items,
+           remove.failures, remove.vars.DIF.missing, remove.vars.DIF.constant,
+           verbose),
+         checkmate::assert_logical, len = 1)
+
+  # arguments specific to `conquest`:
+  if(software == "conquest"){
+    # character: dir, conquest.folder, model.statement, export
+    checkmate::assert_character(dir, len = 1)
+    checkmate::assert_character(model.statement, len = 1, null.ok = TRUE)
+    checkmate::assert_character(export)
```

Comment:
I think, `export` must be a list of several logical arguments, not a character vector